README.md

```diff
@@ -0,0 +1,180 @@
+  is fully populated to send to a downloader as the content-length header, based on the 
+  variable, dimension, and record specifications. No actual data needed.
+  - `close()` - close the file pointer, if there is one.
+  - `createDimension()` and `createVariable()` - add dimensions or variables to the file
+  - `flush()` - flush to disk. Only relevant for files on the filesystem
+  - `recvars()` and `nonrecvars()` - get an ordered dictionary of the record or nonrecord 
+    variables in this file
+  - `__generate__()` - get this file as generator chunks
+  - `_write()` - write the netCDF file to disk. Only relevant for files on the filesystem
+  - `_header()`, `_data()`, `_dim_array()`, `_gatt_array()`, `_var_array()`, `_att_array()`, `_num_recs()`, `_var_metadata()`
+    get the various parts of a netCDF file, encoded per the netCDF specification. Helper
+    functions to `_write()` and `__generate__()`. Reading the netCDF spec is helpful for
+    understanding the use of each of these functions; their names match up.
+  - `_read()` - read a netCDF file from the filesystem
+  - `_read_values()`, `_read_dim_array()`, `read_gatt_array()`, `read_var_array()`, `_read_att_array()`, `_read_num_recs()`, `_read_var_values()`
+    read part of a netCDF file from the filesystem, following the netCDF spec. Helper functions
+    to _read(). Reading the netCDF spec is helpful for understanding the use of each of 
+    these functions; their names match up. Not used by the PDP. 
+  - `_calc_begins()` - calculates the starting address of each variable. Essential for
+    streaming data if you want to send the header before you have all the variable data.
+  - `set_numrecs()` - set the number of records in the datafile without actually assigning the
+    data. Used for streaming data to send the filesize and complete headers before all
+    data is available.
+  - `_pack_begin()`, `_pack_int()`, `_pack_int_64()`, `_pack_string()`
+    return an integer or string in the correct format for a netCDF file. Helpers to
+    the file-writing or file-generator functions. NetCDF is big-endian
+  - `_unpack_int()`, `_unpack_int64()`, `unpack_string()`
+    read the file and translate an integer or string. Helpers to the _read functions
+  PDP Usage: To create headers for streaming, the PDP (via pydap.responses.netcdf) creates
+  a new, virtual netCDF file with the dimensions and variables requested by the user, and
+  global and variable attributes read from the source netCDF file. Then the number of
+  records is set and the variable begin addresses are calculated. Then the header can
+  be generated and streamed to the downloader. A pydap handler package streams the data
+  from the source file; it is never added to this virtual netcdf file used to generate
+  the headers. A NcOrderedDict is used to synchronize the order of the variables in
+  the header generated here and the data generated by pydap.handlers.hd5.
```

Comment:
This very useful paragraph is too big to just run on after the definition. Suggest making it a sub-list item to set it off.